Assets/Mirror/Components/NetworkTransformSnapshotInterpolation.cs

```diff
@@ -0,0 +1,351 @@
+            _nextSyncInterval = Time.time + interval;
+            lastPosition = Position;
+            lastRotation = Rotation;
+        }
+
+        /// <summary>
+        /// Has target moved since we last checked
+        /// </summary>
+        /// <returns></returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        bool HasMoved()
+        {
+            bool moved = Vector3.Distance(lastPosition, Position) > positionSensitivity;
+
+            if (moved)
+            {
+                lastPosition = Position;
+            }
+            return moved;
+        }
+
+        /// <summary>
+        /// Has target moved since we last checked
+        /// </summary>
+        /// <returns></returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        bool HasRotated()
+        {
+            bool rotated = Quaternion.Angle(lastRotation, Rotation) > rotationSensitivity;
+
+            if (rotated)
+            {
+                lastRotation = Rotation;
+            }
+            return rotated;
+        }
+
+        void Update()
+        {
+            if (isServer)
+            {
+                ServerSyncUpdate();
+            }
+
+            if (isClient)
+            {
+                if (IsLocalClientInControl)
+                {
+                    ClientAuthorityUpdate();
+                }
+                else
+                {
+                    ClientInterpolation();
+                }
+            }
+        }
+
+        #region Server Sync Update
+        void ServerSyncUpdate()
+        {
+            if (ServerNeedsToSendUpdate())
+            {
+                SendMessageToClient();
+                ClearNeedsUpdate(clientSyncInterval);
+            }
+        }
+
+        /// <summary>
+        /// Checks if object needs syncing to clients
+        /// <para>Called on server</para>
+        /// </summary>
+        /// <returns></returns>
+        bool ServerNeedsToSendUpdate()
+        {
+            if (IsControlledByServer)
+            {
+                return IsTimeToUpdate() && (HasMoved() || HasRotated());
+            }
+            else
+            {
+                // dont care about time here, if client authority has sent snapshot then always relay it to other clients
+                // todo do we need a check for attackers sending too many snapshots?
+                return _needsUpdate;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        void SendMessageToClient()
+        {
+            // if client has send update, then we want to use the state it gave us
+            // this is to make sure we are not sending host's interpolations position as the snapshot insteading sending the client auth snapshot
+            TransformState state = IsControlledByServer ? TransformState : _latestState;
+            // todo is this correct time?
+            RpcServerSync(state, NetworkTime.time);
+        }
+
+        [ClientRpc]
```

Comment:
```suggestion
        [ClientRpc(channel = Channels.DefaultUnreliable)]
```
Default unreliable for transports that support it.